crates/astria-sequencer/src/app/mod.rs

```diff
@@ -162,9 +162,13 @@ pub(crate) struct App {
     // to the mempool to recost all transactions.
     recost_mempool: bool,
 
-    // cache of results of executing of transactions in `prepare_proposal` or `process_proposal`.
-    // cleared at the end of each block.
-    execution_results: Option<Vec<tendermint::abci::types::ExecTxResult>>,
+    // cache of results of executing of transactions in `prepare_proposal`.
+    // cleared in `process_proposal` if we're the proposer.
+    execution_results: Option<Vec<ExecTxResult>>,
+
+    // cache of results of executing of transactions in `process_proposal`.
+    // cleared at the end of the block.
+    finalize_block: Option<abci::response::FinalizeBlock>,
```

Comment:
Should we utilize the ephemeral storage within cnidarium vs adding to state on the app? I know @SuperFluffy has been trying to decrease our global state bloat.

Reply:
this is functionally the same - it's an object stored in memory and cleared at the start of each block. the ephemeral store is also only in memory and is cleared at the end of each block

Reply:
I agree that we should employ the ephemeral store more rather than adding fields to app.

There is a tendency to add more and more fields to `App` that need to be actively set/unset, which is not clean (and I believe with which we had issues in the past).

For a reader it's also better to have a single source of truth of effects that happen during block building, rather than finding some in the ephemeral store and some in the "global" app state.

Reply:
moved the fields to the ephemeral object store here: https://github.com/astriaorg/astria/pull/1562/commits/591fd94c89ae27b26c5b067f423dd2a657f87d41

this is nice because the fields are automatically cleared when `update_state_for_new_round` is called, however having to get a `StateDelta` for writing to it is bit more cumbersome in parts (putting the `execution_results` in `prepare_proposal`)